### PR TITLE
Add support for multiple nack messages in deli

### PR DESCRIPTION
**Pull request by GaryWilber**
Deli's `nackMessages` system currently supports 1 nack messages type at a time and it's currently used to set/unset the SummaryMaxOps type.

I need the ability to set more than one at a time. This means it needs to track each nack messages type separately. So I'm changing nackMessages to a Map. It will keep track of active nack messages that are enabled per nack message type.
The NackMessages control messages will allow arbitrary control of enabling/disabling nack message types.

My use case involves preventing deli from sequencing ops when there are certain backend things occurring, such as an SPO tenant/site/file being read-only. And I don't want that nack message type to interfere with the SummaryMaxOps nack message type. Related to issue #9000

### CODE REVIEW

File: server/routerlicious/packages/lambdas/src/deli/lambda.ts [hunk 1]
```diff
@@ -478,7 +497,7 @@ export class DeliLambda extends TypedEventEmitter<IDeliLambdaEvents> implements
             this.sessionMetric,
             this.sequenceNumber,
             this.durableSequenceNumber,
-            this.nackMessages?.identifier,
+            this.nackMessages.size > 0 ? Array.from(this.nackMessages.keys())[0] : undefined,
```
Comment (hedasilv) [inline server/routerlicious/packages/lambdas/src/deli/lambda.ts:500]: `logCommonSessionEndMetrics` last argument is `timeoutErrorReason`. If I understand it correctly, we are passing the first "identifier" in the `nackMessages` map as the reason. Could it be possible, if `nackMessages` size is > 1, that there is more than one `timeoutErrorReason`? Or maybe that the reason be something other than the element at position 0 of the array created above?

File: server/routerlicious/packages/lambdas/src/deli/lambda.ts [hunk 2]
```diff
@@ -493,37 +512,38 @@ export class DeliLambda extends TypedEventEmitter<IDeliLambdaEvents> implements
...
-                if (clientId) {
-                    const client = this.clientSeqManager.get(clientId);
-                    if (client) {
-                        for (const scope of nackMessages.allowedScopes) {
-                            if (client.scopes.includes(scope)) {
-                                // this client has an allowed scope. don't nack it
-                                shouldNack = false;
-                                break;
+        if (this.nackMessages.size > 0 && this.serviceConfiguration.deli.enableNackMessages) {
+            for (const nackMessages of this.nackMessages.values()) {
```
Comment (hedasilv) [inline server/routerlicious/packages/lambdas/src/deli/lambda.ts:516]: Should we rename this to `nackMessageControlMessageContents`? It is a long name, and I know the original code also used `nackMessages`, but reading the code for the first time made me confused between `nackMessages` and `this.nackMessages`. I think renaming might make it easier to understand it.

**Review by hedasilv — APPROVED** [review]

File: server/routerlicious/packages/lambdas/src/deli/lambda.ts (hunk 2 above)
Comment (GaryWilber) [reply server/routerlicious/packages/lambdas/src/deli/lambda.ts:516]: Sounds good. I renamed it

File: server/routerlicious/packages/lambdas/src/deli/lambda.ts (hunk 1 above)
Comment (GaryWilber) [reply server/routerlicious/packages/lambdas/src/deli/lambda.ts]: I fixed this up. It will pass a list now

**Review by hedasilv — APPROVED** [review]

**Pull request merged**